pkg/controller/apigatewayv2/api/setup.go

```diff
@@ -36,30 +36,36 @@ import (
 // SetupAPI adds a controller that reconciles API.
 func SetupAPI(mgr ctrl.Manager, l logging.Logger) error {
 	name := managed.ControllerName(svcapitypes.APIGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.postCreate = postCreate
+			e.preDelete = preDelete
+		},
+	}
 	return ctrl.NewControllerManagedBy(mgr).
 		Named(name).
 		For(&svcapitypes.API{}).
 		Complete(managed.NewReconciler(mgr,
 			resource.ManagedKind(svcapitypes.APIGroupVersionKind),
-			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
 			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
 			managed.WithLogger(l.WithValues("controller", name)),
 			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
 }
 
-func (*external) preObserve(context.Context, *svcapitypes.API) error {
+func preObserve(_ context.Context, cr *svcapitypes.API, obj *svcsdk.GetApiInput) error {
+	obj.ApiId = aws.String(meta.GetExternalName(cr))
 	return nil
 }
-func (*external) postObserve(_ context.Context, cr *svcapitypes.API, _ *svcsdk.GetApiOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+
+func postObserve(_ context.Context, cr *svcapitypes.API, _ *svcsdk.GetApiOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
 	cr.SetConditions(xpv1.Available())
 	return obs, err
 }
```

Comment:
Do we need to check error before setting `Available` here?

Reply:
It does seem like in practice we are always passing `nil` error here, but I see it is checked in other places, so not sure what pattern we want to follow

Reply:
Yeah, good catch!